### PR TITLE
[#158476864] add qrcode custom type

### PR DESCRIPTION
**Pull request by gunzip**
1. renamed CodiceSegregazione -> segregationCode
1. added type for QrCode string
1. added WithinRangeInteger / NonNegativeInteger to handle integers from swagger specs
1. added FiscalCode and OrganizationFiscalCode types

### CODE REVIEW

File: src/pagopa.ts [hunk 1]
```diff
@@ -193,3 +196,73 @@ export const PaymentNoticeNumberFromString = new t.Type<
...
+
+// (6a) "PAGOPA"
+// (3n) "002"
+// (18n) Numero Avviso (PaymentNoticeNumberFromString)
+// (11a) codiceIdentificativoEnte (organizationFiscalCode)
+// (10n) Amount in euro cents (digits)
+
+// ie. "PAGOPA|002|123456789012345678|12345678901|1234567801"
+
+const QrCode2 = t.interface({
```
Comment (cloudify) [inline src/pagopa.ts]: `PaymentNoticeQrCode` ?

Comment (cloudify) [reply src/pagopa.ts]: who knows, we may have other QR codes in the future

File: src/strings.ts [hunk 2]
```diff
@@ -127,3 +127,19 @@ const v6 = `
...
+export const FiscalCode = PatternString(
+  "^[A-Z]{6}[0-9LMNPQRSTUV]{2}[ABCDEHLMPRST][0-9LMNPQRSTUV]{2}[A-Z][0-9LMNPQRSTUV]{3}[A-Z]$"
+);
+
+export type FiscalCode = t.TypeOf<typeof FiscalCode>;
+
+/**
+ * A valid Fiscal Number (for organizations)
+ */
+export const OrganizationFiscalCode = PatternString("^[0-9]{11}$");
```
Comment (cloudify) [inline src/strings.ts]: english name for PIVA is `VatNumber` or `VatId`

Comment (gunzip) [reply src/strings.ts]: then it must be changed here as well:

https://github.com/teamdigitale/digital-citizenship-functions/blob/327442dc144334b5d91adae65be231d4c9a4100f/lib/models/service.ts#L40

**Comment by digitalcitizenship** [comment]
## Affected stories

  * ⚙️ [#158476864](https://www.pivotaltracker.com/story/show/158476864): Aggiungere un tipo custom per la gestione del QrCode

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/strings.ts (hunk 2 above)
Comment (cloudify) [reply src/strings.ts]: But that's a different definition, that's a `FiscalCode` (it follows the fiscal code pattern) - while this one it's all numbers, it's indeed a PIVA (that has a different pattern than CF).

I think the confusion here is caused by the subject, they're both fiscal code (aka tax ID) but:

* the FiscalCode like FRLFRC... is a personal tax identification number
* the PIVA 12347... is a company tax identification number

what we should provide here is the ID format (Codice Fiscale or Partita Iva), then it's up to the user of the type to associate it to the right entity

I think it makes sense to use an already defined terminology (e.g. Partita IVA) than create a new one (e.g. Codice Fiscale dell'Organizzazione)

Comment (gunzip) [reply src/strings.ts]: this one (PIVA) must match the organizationFiscalCode (which is the "Codice Fiscale dell'Organizzazione" that we save into the service metadata). That's why I'm suggesting to rename organizationFiscalCode to PIVA also in the notification APIs *and* to [fix the format](https://www.pivotaltracker.com/story/show/158476749).

**Pull request merged**